### ISSUE TITLE
Warriors should be able to translate kata instructions to other spoken languages.

### ISSUE THREAD

**Issue by camposbotellob** [Codewars/codewars.com#1160]
Currently, warriors can translate kata to other programming languages. However, it would be interesting to expand the site from being "English only" to include support for other spoken languages. As a Spanish speaker myself, I would be very interested in translating kata instructions to my language.

**Comment by Voileexperiments** [comment]
For the record, [according to Alexa](https://www.alexa.com/siteinfo/codewars.com) currently the most dominating demographics of CW users are US users and... Chinese users.

So in the end, we might be seeing Chinese going around everyone on CW.

**Comment by camposbotellob** [comment]
It isn't surprising that a 100% English site is popular in the US and the UK, as well as the few other countries that teach English to their young. I'm just saying that allowing people to translate kata wouldn't exactly hurt anyone, and it would probably make the site popular in more non-English-speaking countries.

**Comment by dinglemouse2250** [comment]
> allowing people to translate kata wouldn't exactly hurt anyone

Wouldn't it? I see nothing but trouble.

---

Kata descriptions are often bad right now with only one language. Imagine that multiplied across languages.

If CW was to allow the Kata description text to be in multiple languages then it introduces increasing risk that translations differ from the original Kata author intent... but the author has no way to tell that.

Furthermore it means the author can no longer edit/improve the Kata description without risk that all the translated descriptions become out-of-sync and would also need updating... but the author probably does not know the other language so cannot make that change unassisted.

Basically, it takes control of the Kata away from the Kata Sensei. I don't see that is a good thing.

It will cause also unnecessary arguments/debates about requirements and solutions - e.g. "It said do this (in English), but that rule said something else (in Spanish)". Or "there was a big hint that was only available in French... ".

What sounds like a good idea has potential to become a big can of worms.

My 2c

**Comment by camposbotellob** [comment]
People who translate kata into other programming languages already have to do this. In order to translate, the translator has to have completed the kata, which implies that he or she has sufficient understanding of the instructions and what is expected. Furthermore, current translations to other programming languages are not just put out there as soon as they are completed, they have to go through an approval process by the community. It is also my understanding that other warriors can offer translations and the community decides on the best one.

**Comment by kazk** [comment]
Thanks for the suggestion, but we're not going add this. Maybe it's possible to have something external that's entirely managed by the community.

Kata descriptions are already out of control and the amount of work required to manage translations is too high.
It's also a huge effort just to translate over 8000 kata descriptions that we have at the moment. So the user experience will be terrible.

> People who translate kata into other programming languages already have to do this.

This is not a fair comparison. Many power users can understand most of the supported programming languages, but I assume most of them only know few natural languages.
